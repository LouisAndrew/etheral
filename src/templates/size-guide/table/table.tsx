import React from 'react';
// import styling libs
import { Box, Text } from 'rebass';
// import local components
import { Data } from '..';

type Props = {
    data: Data[];
};

/**
 * Table component to display available sizes available..
 */
const Table: React.FC<Props> = ({ data }) => {
    const border = '2px solid #222';
    const textAlign: 'center' | 'left' = 'center';

    const baseTextStyle = {
        fontFamily: 'body',
        fontSize: 1,
        px: [4, 4, 6],
        py: [2],
        textAlign,
    };

    const headingStyle = {
        ...baseTextStyle,
        fontWeight: 'bold',
    };

    return (
        <Box
            mb={[6, 6, 0]}
            mr={[0, 0, 4, 9]}
            sx={{
                display: 'grid',
                gridTemplateColumns: 'repeat(3, auto)',
                border,
                borderBottomWidth: 0,
            }}
        >
            {/* table heading! */}
            <Text
                {...headingStyle}
                sx={{
                    gridColumn: '1/2',
                    borderBottom: border,
                }}
            >
                SIZE
            </Text>
            <Text
                {...headingStyle}
                sx={{
                    gridColumn: '2/3',
                    borderLeft: border,
                    borderBottom: border,
                }}
            >
                DIAMEsTER
            </Text>
            <Text
                {...headingStyle}
                sx={{
                    gridColumn: '3/4',
                    borderLeft: border,
                    borderBottom: border,
                }}
            >
                CIRCUMFERENCE
            </Text>

            {/* render data */}
            {data.map(dt => (
                <React.Fragment key={dt.size}>
                    <Text
                        {...baseTextStyle}
                        sx={{
                            gridColumn: '1/2',
                            borderBottom: border,
                        }}
                    >
                        {dt.size}
                    </Text>
                    <Text
                        {...baseTextStyle}
                        sx={{
                            gridColumn: '2/3',
                            borderLeft: border,
                            borderBottom: border,
                        }}
                    >
                        {dt.diameter} mm
                    </Text>
                    <Text
                        {...baseTextStyle}
                        sx={{
                            gridColumn: '3/4',
                            borderLeft: border,
                            borderBottom: border,
                        }}
                    >
                        {dt.circum} mm
                    </Text>
                </React.Fragment>
            ))}
        </Box>
    );
};

export { Table };
