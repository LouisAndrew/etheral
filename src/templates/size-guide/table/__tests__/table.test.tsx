import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

// import { render, cleanup } from '@testing-library/react'
import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';

import { Table } from '../table';

describe('Table', () => {
    const Element = <Table data={[]} />;

    afterEach(cleanup);

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(Element, div);
    });

    /* it('renders correctly', () => {
		const { getByTestId } = render()
	}) */

    it('matches snapshot', () => {
        const run = true;

        if (run) {
            const tree = renderer.create(Element).toJSON();
            expect(tree).toMatchSnapshot();
        }
    });
});
