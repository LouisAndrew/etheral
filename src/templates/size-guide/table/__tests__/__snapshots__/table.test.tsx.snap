// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table matches snapshot 1`] = `
<div
  className="css-1119h9n"
>
  <div
    className="css-1ps274q"
  >
    SIZE
  </div>
  <div
    className="css-wv9yu9"
  >
    DIAMATER
  </div>
  <div
    className="css-mo0fek"
  >
    CIRCUMFERENCE
  </div>
</div>
`;
